qiskit/circuit/equivalence.py

```diff
@@ -22,9 +23,6 @@
 from .parameterexpression import ParameterExpression
 
 Key = namedtuple("Key", ["name", "num_qubits"])
-
-Entry = namedtuple("Entry", ["search_base", "equivalences"])
-
```

Comment:
Is it possible to keep using a structured object like this `namedtuple` instead of a `dict` with Rustworkx?  It's generally much easier to reason about and modify the code when it's using an object with a code-enforced structure rather than ad-hoc dictionaries.

Reply:
There is a `named_tuple!` macro available in rust, but I don't know how or where the translation would take place between the stored node and edge weights from python to rust. I'll let @mtreinish comment on this.

Reply:
Rustworkx doesn't put any constraints on the payloads you attach to a graph. You can put any python object as the payload for a node or edge, so it can be a `namedtuple`, a `dict`, or anything else. But, you can only access the nodes of a graph as a mapping based on index though, so something like `graph[index] = namedtuple()`. I assume @jakelishman is just asking to make the node payload a `namedtuple` instead of `{"key": key, "equivs": []}`  which should work fine without any issues.

Reply:
All dicts have been converted to namedtuples in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/73b553615a6b84098b67a6f53daa7c84f4cce872.